app/v2/controllers/V2ArrivalsController.scala

```diff
@@ -79,14 +81,29 @@ class V2ArrivalsControllerImpl @Inject() (
         (for {
           _ <- validationService.validateXml(MessageType.ArrivalNotification, request.body).asPresentation
           _ = auditService.audit(AuditType.ArrivalNotification, request.body, MimeTypes.XML)
-
           arrivalNotificationResult <- persistAndSend(request.body)
         } yield arrivalNotificationResult).fold[Result](
           presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
           result => Accepted(HateoasArrivalNotificationResponse(result.arrivalId))
         )
     }
 
+  def createArrivalNotificationJSON(): Action[Source[ByteString, _]] =
+    (authActionNewEnrolmentOnly andThen messageSizeAction()).stream {
+      implicit request =>
+        implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+
+        (for {
+          _ <- validationService.validateJson(MessageType.ArrivalNotification, request.body).asPresentation
+          _ = auditService.audit(AuditType.ArrivalNotification, request.body, MimeTypes.JSON)
+          xmlSouce                  <- conversionService.jsonToXml(MessageType.ArrivalNotification, request.body).asPresentation
+          arrivalNotificationResult <- persistAndSend(xmlSouce)
```

Comment:
I'm not sure this'll work as you think, because the `xmlSource` will be a one-use source, a bit like the request stream - we'll need this to use `withReusableSource` like Rob does in his PR.

Reply:
https://github.com/hmrc/common-transit-convention-traders/pull/317/files#diff-4530c97418bd3c5db8f389bc12c415848247c437e9393aa9b85dc562227e656dR252-R262 is one way to do it.

Reply:
For now we only need it once right? I thought it would be something to refactor when PPNS is added?

Reply:
Oh of course! My bad...